sensu/resource_filter.go

```diff
@@ -56,7 +56,13 @@ func resourceFilterCreate(d *schema.ResourceData, meta interface{}) error {
 
 	expressions := expandStringList(d.Get("expressions").([]interface{}))
 	runtimeAssets := expandStringList(d.Get("runtime_assets").([]interface{}))
-	when := expandTimeWindows(d.Get("when").(*schema.Set).List())
+	// to prevent the creation of an empty when block
+	whenRaw, whenExist := d.GetOk("when")
+	expandedTimeWindows := expandTimeWindows(whenRaw.(*schema.Set).List())
```

Comment:
why not just
```go
when := expandTimeWindows(whenRaw.(*schema.Set).List())
```
line 62 just seems unnecessary.

Reply:
the `EventFilter` later takes a `when` pointer, so we need to pass an object address. However, we cannot get address for nil object. So we are trying to get a value which is either the address or nil here. other things we noticed, we can’t get the address of a function return value without creating a temporary variable, so we created this temp var. This is why we have this weird looking logic: trying to determine the address of the return value or assign nil to the variable.